include/aie/Dialect/AIEVec/IR/AIEVecOps.td

```diff
@@ -313,16 +313,33 @@ def AIEVec_MulOp:
 
 def AIEVec_MulElemOp:
   AIEVec_Op<"mul_elem", [
-    Pure
+    Pure, 
+    IsValidAIE2MulElemShapeAndType<"lhs", "rhs", "acc">
   ]>,
-  Arguments<(ins AnyVector:$lhs, AnyVector:$rhs)>,
-  Results<(outs AnyVector:$result)> {
+  Arguments<(ins AnyVector:$lhs,
+                 AnyVector:$rhs)>,
+  Results<(outs AnyVector:$acc)> {
```

Comment:
We don't support `AnyVector`, you should use an operand type constraint for these, and then your op constraint will be simpler.